### ISSUE TITLE
Injecting proc arguments

### ISSUE THREAD

**Issue by dawkot** [nim-lang/Nim#12355]
### Summary
It would be cool if you could inject arguments in a more terse way

### Description

Currently you have to inject proc arguments like this:
```nim
template foo(impl) =
  proc _(x: int) =
    let x {.inject.} = x
    impl

foo echo x
```

I would this to work
```nim
template foo(impl) =
  proc _(x {.inject.}: int) = impl

foo echo x
```

### Alternatives

Wouldn't be as useful if this was fixed https://github.com/nim-lang/Nim/issues/10609

**Comment by Araq** [comment]
You can also use a ``.dirty`` template:

```nim

template foo(impl) {.dirty} =
  proc _(x: int) =
    impl

foo echo x
```

**Comment by awr1** [comment]
Pragmas don't seem to work at all for arguments. For instance, this fails:

```nim
template foo {.pragma.}
proc x(y {.foo.}: int) = discard
```

**Comment by Araq** [comment]
Pragmas for *parameters* do not exist yet, yes. We need them...

**Comment by ghost** [comment]
This now works:
```nim
template foo(name, impl: untyped): untyped =
  proc name(x {.inject.}: int) = impl

foo(test):
  echo x

test(5)
```

Can we close the issue or maybe we should document this feature first?

**Comment by Clyybber** [comment]
Fixed with test in #15178
